trillian/ctfe/handlers.go

```diff
@@ -476,22 +476,18 @@ func addChainInternal(ctx context.Context, li *logInfo, w http.ResponseWriter, r
 	}
 
 	glog.V(2).Infof("%s: %s => grpc.QueueLeaves", li.LogPrefix, method)
-	rsp, err := li.rpcClient.QueueLeaves(ctx, &req)
+	rsp, err := li.rpcClient.QueueLeaf(ctx, &req)
```

Comment:
what does QueueLeaf return? specifically, could it return a resp.QueueLeaf = nil? ref here and L487.

Reply:
It returns a `QueueLeafResponse`: https://github.com/google/trillian/blob/master/trillian_log_api.proto#L214

Things would be very wrong if it returned a nil QueueLeaf, but I'll put something to handle that case, to be safe.

Reply:
Yes, indeed. I'm guessing there's a check in QueueLeaf to guard for that and it would return err != nil. Thanks for the extra safe belts and braces.